draft-regext-rdap-extensions.md

```diff
@@ -386,125 +484,187 @@ that example could be written as:
+extension identifiers in the "rdapConformance" member of an RDAP
+response:
+
+    A response to a "help" request will include identifiers for all of
+    the specifications supported by the server. A response to any
+    other request will include only identifiers for the specifications
+    used in the construction of the response.
+
+A strict interpretation of this wording where "construction of the
+response" refers to the JSON structure only would rule out the use of
+(#profiles_and_markers) extension identifiers, which are in common use
+in RDAP.  This document updates the guidance. For responses to queries
+other than "/help", a response MUST include in the "rdapConformance"
+array only those extension identifiers necessary for a client to
+deserialize the JSON and understand the semantic meaning of the
+content within the JSON, and each extension identifier MUST be free
+from conflict with the other identifiers with respect to their syntax
+and semantics.
+
+Note that this document does not update the guidance from [@!RFC9083,
+Section 4.1] regarding "/help" responses and the "rdapConformance"
+array.
+
+### Camel Casing {#camel_casing}
+
+The styling convention used in [@!RFC9083] for JSON names is often
+called "camel casing", in reference to the hump of a camel. In this
+style, the first letter of every word, except the first word,
+composing a name is capitalized.  This convention was adopted to
+visually separate the namespace from the name, with an underscore
+between them.  Though there is no explicit guidance to use camel case
+names, extensions would be wise to continue the style.
+
+## Identifier Omission {#identifier_omission}
+
+Though all RDAP extensions are to be registered in the IANA RDAP
+extensions registry, there is an implicit two-class system of
+extensions that comes from the ownership of the RDAP specifications by
+the IETF: extensions created by the IETF and extensions not created by
+the IETF.
+
+In the perspective of how extension identifiers are used as namespace
+separators, extensions created by the IETF are not required to use the
+extension identifier as a prefix in requests and responses, as the
+IETF can coordinate its own activities to avoid name collisions. In
+practice, most extensions owned by the IETF do use extension
+identifiers as prefixes in their requests and responses.
+
+RDAP extensions not defined by the IETF MUST use the extension
+identifier as a prefix in accordance with this document, [@!RFC7480],
+[@!RFC9082], and [@!RFC9083].  RDAP extensions defined by the IETF
+SHOULD use the extension identifier as a prefix or as a bare extension
+identifier (see (#bare_extension)).  IETF-defined RDAP extensions that
+do not follow this guidance MUST describe why it is not being
+followed.
 
 In addition, RDAP extensions defined by the IETF are allowed to define
-new types in the RDAP JSON Values Registry (see (#rdap_json_values_registry)).
+new types in the RDAP JSON Values Registry (see
```

Comment:
This sentence seems to be out of place in this section, no?

Reply:
Yep, mainly due to the new title for this section.  Having said that, I'm not sure the text adds that much anyway, so what about just removing it?

Reply:
IIRC, this was specifically discussed in the WG. We would need to find a more appropriate section for it. :)

Reply:
But non-IETF-defined extensions can request new values in this registry as well, since the policy for the registry is 'expert review'.  There are values in the registry from the ICANN and NRO profiles, for example.

Reply:
They can register new values but not new types.

Reply:
Ah yep, thanks.  The "RDAP JSON Values Registry" section has the following at the moment:

```
In addition to the registration of values, RDAP extensions defined by
the IETF and other IETF specifications MAY define additional value
types (the "type" field).  These specifications MUST describe the
specific JSON field to be used for each new value type.
```

so I'll remove the commented-on text here, unless anybody has any concerns about that.